src/templates/TemplateData.ts

```diff
@@ -133,107 +133,35 @@ function verifyTemplatesByRuntime(templatesMap: { [runtime: string]: Template[]
+export async function tryGetTemplateData(reporter: TelemetryReporter | undefined, cliFeedJson: cliFeedJsonResponse, globalState?: vscode.Memento, isBackup: boolean = false): Promise<TemplateData | undefined> {
+    const templateReleaseVersion: string = 'templateReleaseVersion';
+    const v1ReleaseVersion: string = '1.0.12'; // known stable version
+    const betaReleaseVersion: string = '2.0.1-beta.25'; // known stable version
     try {
         return <TemplateData>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetLatestTemplateData', reporter, undefined, async function (this: IActionContext): Promise<TemplateData> {
             this.suppressErrorDisplay = true;
             this.properties.isActivationEvent = 'true';
             const templatesMap: { [runtime: string]: Template[] } = {};
             const configMap: { [runtime: string]: Config } = {};
             for (const key of Object.keys(ProjectRuntime)) {
+                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
                 // called within loop in case setting has changed between runtimes
                 const userTemplateVersion: string | undefined = getFuncExtensionSetting(templateVersionSetting);
-                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
                 const feedRuntime: string = getFeedRuntime(runtime);
-                const releaseVersion: string = userTemplateVersion ? userTemplateVersion : <string>cliFeedJson.tags[feedRuntime].release;
-                await downloadAndExtractTemplates(cliFeedJson, releaseVersion);
-
-                // only Resources.json has a capital letter
-                const rawResources: object = <object>await fse.readJSON(path.join(tempPath, 'resources', 'Resources.json'));
-                const rawTemplates: object[] = <object[]>await fse.readJSON(path.join(tempPath, 'templates', 'templates.json'));
-                const rawConfig: object = <object>await fse.readJSON(path.join(tempPath, 'bindings', 'bindings.json'));
-
-                [templatesMap[runtime], configMap[runtime]] = parseTemplates(rawResources, rawTemplates, rawConfig);
-                verifyTemplatesByRuntime(templatesMap, runtime);
-                if (globalState) {
-                    globalState.update(`${runtime}-release`, releaseVersion);
-                    globalState.update(getRuntimeKey(templatesKey, runtime), rawTemplates);
-                    globalState.update(getRuntimeKey(configKey, runtime), rawConfig);
-                    globalState.update(getRuntimeKey(resourcesKey, runtime), rawResources);
+                let releaseVersion: string = userTemplateVersion ? userTemplateVersion : cliFeedJson.tags[feedRuntime].release;
+                if (isBackup) {
+                    releaseVersion = runtime === ProjectRuntime.one ? v1ReleaseVersion : betaReleaseVersion;
```

Comment:
I think you originally said you would pass in 'templateVersion' to tryGetTemplateData, which I like better than 'isBackup'

Reply:
I did say that, but when I tried to implement it, I'd have to pass both templateVersions, one for each runtime as a string[] or call the function for each runtime.

Reply:
"call the function for each runtime" sounds the best to me